### PR TITLE
add singer embedding on RNN, GLU Transformer & Confofmer

### PR DESCRIPTION
**Pull request by PeterGuoRuc**

### CODE REVIEW

File: SVS/model/infer.py [hunk 1]
```diff
@@ -140,51 +177,99 @@ def infer(args):
...
-            dec_pos_enc_layer_type=args.dec_pos_enc_layer_type,
-            dec_selfattention_layer_type=args.dec_selfattention_layer_type,
-            dec_activation_type=args.dec_activation_type,
-            dec_use_cnn_module=args.dec_use_cnn_module,
-            dec_cnn_module_kernel=args.dec_cnn_module_kernel,
-            dec_padding_idx=args.dec_padding_idx,
-            device=device,
-        )
+        if args.db_joint:
+            model = ConformerSVS_FULL_combine(
```
Comment (ftshijt) [inline SVS/model/infer.py:181]: I recommend using a new name (this name is not informative). Though we are in new phase of reformating code, it is still important to make the name clear.

File: SVS/model/infer.py [hunk 2]
```diff
@@ -225,30 +310,56 @@ def infer(args):
...
-        sing_quality=args.sing_quality,
-    )
-    collate_fn_svs = SVSCollator(
-        args.num_frames,
-        args.char_max_len,
-        args.use_asr_post,
-        args.phone_size,
-    )
+    if args.db_joint:
+        test_set = SVSDataset_combine(
```
Comment (ftshijt) [inline SVS/model/infer.py:322]: could you combine the SVSdataset and SVSDataset_combine together (e.g., with some arguments)

File: SVS/model/network.py [hunk 3]
```diff
@@ -470,6 +470,82 @@ def forward(self, encoder_out, align_phone, text_phone, pitch, beats):
...
+        self.emb_singer = nn.Embedding(singer_size, embed_size)
+        self.fc_out = nn.Linear(2 * embed_size, embed_size)
+
+    def aligner(self, encoder_out, align_phone, text_phone):
+        """aligner."""
+        # align_phone = [batch_size, align_phone_length]
+        # text_phone = [batch_size, text_phone_length]
+        # align_phone_length( = frame_num) > text_phone_length
+        # batch
+        align_phone = align_phone.long()
```
Comment (ftshijt) [inline SVS/model/network.py:496]: how about constructing a function here to uniform the alignment (I believe they are the same for all models)

**Comment by ftshijt** [comment]
Also, please pass the CI, thanks!

**Comment by ftshijt** [comment]
- Fix CI
- Resolve the comments

Then, I will merge it

File: SVS/model/infer.py (hunk 2 above)
Comment (PeterGuoRuc) [reply SVS/model/infer.py:322]: I will fix it

File: SVS/model/network.py (hunk 3 above)
Comment (PeterGuoRuc) [reply SVS/model/network.py:496]: I will fix it.

**Pull request merged**